artifacts/deployment/etcdproxy-controller.yaml

```diff
@@ -0,0 +1,158 @@
+---
+# The Namespace where controller and controller resources are deployed.
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: kube-apiserver-storage
+spec:
+---
+# The ServiceAccount used by EtcdProxyController.
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: etcdproxy-controller-sa
+  namespace: kube-apiserver-storage
+---
+# The ServiceAccount used by EtcdProxy pods.
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: etcdproxy-sa
+  namespace: kube-apiserver-storage
+---
+# ClusterRole for etcdproxy-controller-sa to update status on EtcdStorage.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: etcdproxy-crd-clusterrole
+rules:
+- apiGroups: ["etcd.xmudrii.com"]
+  resources: ["etcdstorages"]
+  verbs: ["get", "watch", "list"]
+- apiGroups: ["etcd.xmudrii.com"]
+  resources: ["etcdstorages/status"]
+  verbs: ["get", "watch", "list", "create", "update", "patch"]
+---
+# Role for etcdproxy-controller-sa to manage ReplicaSets, Services, ConfigMap and Secrets.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: etcdproxy-controller-role
+  namespace: kube-apiserver-storage
+rules:
+- apiGroups: ["apps"]
+  resources: ["replicasets"]
+  verbs: ["get", "watch", "list", "create", "update", "patch", "delete"]
+- apiGroups: [""]
+  resources: ["services"]
+  verbs: ["get", "watch", "list", "create", "update", "patch", "delete"]
+- apiGroups: [""]
+  resources: ["secrets", "configmaps"]
+  verbs: ["get", "watch", "list", "create", "update", "patch", "delete"]
+---
+# Role for etcdproxy-sa to read the core etcd Secrets and ConfigMaps.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: etcdproxy-role
```

Comment:
I didn't expect this.  Shouldn't you be mounting those secrets in?

Reply:
Yes, we're mounting the secret in. I'm going to check do we need this rule.

The potential question that I have is: if we want to mount the Secret, does the ServiceAccount that's used by the pod needs to have RBAC rules for that secret?

Reply:
Yes, we're mounting secrets in. Fixed.